### ISSUE TITLE
ResultSets

### ISSUE THREAD

**Issue by thekid** [xp-framework/rfc#309]
# Scope of Change

This RFC suggests changing `rdbms.DBConnection::query()` to always return ResultSet instances, and a method `isSuccess()` to distinguish results from rowsets.
# Rationale

Currently, the method returns a `rdbms.ResultSet` instance _or_ a boolean. This makes nasty instanceof ... handling necessary in any calling code.
# Functionality

``` php
$result= $conn->query('...');
if ($result->isSuccess()) {
  Console::writeLinef('OK, %d row(s) affected', $result->affected());
} else {
  foreach ($result as $row) {
    // ...
  }
}
```
# Security considerations

None.
# Speed impact

Minimalistically slower when using query() because it always creates `ResultSet` instances.
# Dependencies

BC break!
# Related documents

**Comment by thekid** [comment]
> BC break!

[This is an example](https://gist.github.com/thekid/0ca8706e1554d4f26508/8cc7fb8f99a0ce449aee2052520e9bb3576ac804#file-query-class-php-L73): The instanceof check there would need to be replaced by `$result->isSuccess()`.

@kiesel I've verified by checking our corporate codebase for uses of `query()` in conjunction with non-select statements. The five occurrences I've found don't use the return value or test if for thruthiness (a PHP 4 leftover, I guess), thus no real-life breakage occurs as far as I see.
